File: backend/app/utils/sentry.py
import logging

import sentry_sdk
from sentry_sdk import utils
from sentry_sdk.integrations import celery

log = logging.getLogger(__name__)


def init_sentry(dsn: str) -> None:
    try:
        sentry_sdk.init(
            dsn=dsn,
            integrations=[celery.CeleryIntegration()],
            traces_sample_rate=1.0,
            send_default_pii=True,
        )
    except utils.BadDsn as e:
        log.warning("Could not init Sentry: %s", e)

File: frontend/src/services/sentry.ts
import * as Sentry from '@sentry/react';
import { BrowserTracing } from '@sentry/tracing';

export const initSentry = (dsn: string) => {
  try {
    Sentry.init({
      dsn: dsn,
      integrations: [new BrowserTracing()],
      tracesSampleRate: 1.0,
    });
  } catch (e) {
    console.warn(`Could not init Sentry: ${e}`);
  }
};
